### DISCUSSION TITLE
set custom icon for my custom widget (newb)

### DISCUSSION THREAD

**Discussion by daleasberry-n** [eclipse-theia/theia#14680] [Q&A] [answered]
I have put boxes.svg into the src/browser/style directory with index.css. It has the entry:
.playground-icon-class {
    background-image: url('playground.svg');
    background-size: contain;
    width: 16px;
    height: 16px;
    display: inline-block;
}

in my playground-widget.ts I have:
    protected async doInit(): Promise <void> {
        this.id = PlaygroundWidget.ID;
        this.title.label = PlaygroundWidget.LABEL;
        this.title.caption = PlaygroundWidget.LABEL;
        this.title.closable = true;
        this.title.iconClass = 'playground-icon-class';
        this.update();
    }

When I run Theia, the widget is displayed correctly except for the icon. I don't even think the build is picking up the svg. What am I missing?

**Comment by msujew** [comment]
Hey @daleasberry-n,

Does the build pick up your css at all? Do you have an import in your TypeScript code like this:

```ts
import "../../src/browser/style/index.css";
```

**Reply by daleasberry-n** [reply]
Looks like it's not provided out of the box in theia's default webpack.config.js. Since I'm relatively new to Node programming, I didn't know I needed to look there and update it.

**Reply by msujew** [reply]
It's not (directly) part of the webpack config. You just need to import the css for it to be included in your typescript bundle. See:

https://github.com/eclipse-theia/theia/blob/02f4192a044cd9a5fba5e1e7fca1459025d250cb/examples/api-samples/src/browser/api-samples-frontend-module.ts#L27

**Reply by daleasberry-n** [reply, marked as answer]
Found the answer in the css:
.unclosable-window-icon {
  -**webkit-mask**: url("window-icon.svg");
  **mask**: url("window-icon.svg");
}

I was using
 **background-image**: url("window-icon.svg");

**Comment by daleasberry-n** [comment]
Is there an example project somewhere? Can't find in the documentation how to do it and CoPilot is just going in circles.
